packages/website/docs/using-vue.md

```diff
@@ -0,0 +1,204 @@
+import { autocomplete } from "@algolia/autocomplete-js";
+import { createLocalStorageRecentSearchesPlugin } from "@algolia/autocomplete-plugin-recent-searches";
+
+import "@algolia/autocomplete-theme-classic";
+
+const recentSearchesPlugin = createLocalStorageRecentSearchesPlugin({
+  key: "search",
+  limit: 3,
+});
+
+export default {
+  name: "App",
+  setup() {
+    onMounted(() => {
+      autocomplete({
+        container: "#autocomplete",
+        openOnFocus: true,
+        plugins: [recentSearchesPlugin],
+        renderer: {
+          createElement: h,
+          Fragment,
+        },
+        render({ children }, root) {
+          render(children, root);
+        },
+      });
+    });
+  },
+};
+</script>
+```
+
+## Customizing templates
+
+This guide uses the recent searches plugin, which takes care of the display [templates](templates). If you're using the highlighting and snippeting utilities, there's one thing to keep in mind: you must pass them Vue's `createElement` function. Without doing this, the utilities default to `preact.createElement` and won't work properly.
+
+The highlighting and snippeting utilities are:
+- [`highlightHit`](highlighthit)
+- [`snippetHit`](snippethit)
+- [`reverseHighlightHit`](reversehighlighthit)
+- [`reverseSnippetHit`](reversesnippethit)
+
+Here's an example of a custom `item` template using [`snippetHit`](snippethit):
+
+```html title="App.vue"
+<script>
+import { h, Fragment, render, onMounted } from "vue";
+import { autocomplete, snippetHit } from "@algolia/autocomplete-js";
+
+export default {
+  name: "App",
+  setup() {
+    onMounted(() => {
+      autocomplete({
+        // ...
+        getSources({ query }) {
+          return [
+            {
+              // ...
+              templates: {
+                item({ item }) {
+                  return (
+                    <Fragment>
+                      <div className="aa-ItemContent">
+                        <div className="aa-ItemContentTitle">
+                          {snippetHit({
+                            hit: item,
+                            attribute: "name",
+                            createElement: h,
+                          })}
+                        </div>
+                        <div className="aa-ItemContentDescription">
+                          {snippetHit({
+                            hit: item,
+                            attribute: "description",
+                            createElement: h,
+                          })}
+                        </div>
+                      </div>
+                    </Fragment>
+                  );
+                },
+              },
+            },
+          ];
+        },
+        renderer: {
+          createElement: h,
+          Fragment,
+        },
+        render({ children }, root) {
+          render(children, root);
+        },
+      });
+    });
+  },
+};
+</script>
+```
+
+Keep in mind that you should use JSX syntax for your templates.
```

Comment:
I left out the [JSX caveat](https://algolia.slack.com/archives/C01GH87N4CW/p1612908123034100?thread_ts=1612907394.033700&cid=C01GH87N4CW) on React because I assumed it was understood. Do you think it's necessary here, @francoischalifour ?

Reply:
That's good this way!